app/src/main/java/com/example/dungeontest/NamingScreen.kt

```diff
@@ -150,13 +168,24 @@ fun NamingScreen(drawerState: DrawerState, scope: CoroutineScope, base64EncodedP
                     onClick = {
                         scope.launch {
                             val mapName = mapInputNameVal.value.text
-                            viewModel.updateMap(MapRecord(
-                                mapName,
-                                photoUri,
-                            ))
+                            val file = File(photoPath)
+                            val imageBytes = file.readBytes()
+                            val imagePath = saveImageToInternalStorage(context = context, imageBytes = imageBytes, mapName = mapName)
+                            Log.v("NamingScreen", imagePath.toString())
+                            if (imagePath != null) {
+                                val newMap = MapRecord(
+                                    mapName,
+                                    imagePath.toString(),
+                                    null
+                                )
+                                val didSaveMap = viewModel.updateMap(newMap)
+                                viewModel.transitoryMapRecord = newMap
+                            } else {
+                                snackbarHostState.showSnackbar("Failed to save image.")
+                            }
                         }
                         showDialog.value = false
-                        navController.navigate("MainScreen")
+                        navController.navigate("EditorScreen")
```

Comment:
why are we kicking off the navigation here, outside the separately-launched coroutine? Wouldn't we want to wait on starting the navigation step until the stuff currently inside that coroutine was done? Amongh other things, that would be more in line with what was done above for a new/non-overwriting name choice